chinglee-iot/FreeRTOS-Kernel
# Update coverity violation for SMP

Fix coverity violations for SMP diff part.

Description
-----------

Fix coverity violations for SMP diff part.
SMP diff part is generated by comparing the coverity report of **single core** and **SMP**.

The coverity violations with SMP are already fixed in the implementation. The coverify report only shows violations in single core ( common code ). Single core FreeRTOS are checked by another tool "Lint". The reason for violation is documented with lint suppression comment.

The following is a lint suppression example:
```C
/*lint !e9079 All values returned by pvPortMalloc() have at least the alignment required 
 *by the MCU's stack and this allocation allocates a struct that has the alignment requirements of a pointer. */
```

The tasks.c coverity errors for single core


Checker | Violations | Category
-- | -- | --
MISRA C-2012 Directive 4.6 | 7 | Advisory
MISRA C-2012 Rule 10.1 | 3 | Required
MISRA C-2012 Rule 10.3 | 4 | Required
MISRA C-2012 Rule 10.4 | 51 | Required
MISRA C-2012 Rule 11.3 | 3 | Required
MISRA C-2012 Rule 11.4 | 2 | Advisory
MISRA C-2012 Rule 13.2 | 2 | Required
MISRA C-2012 Rule 14.4 | 37 | Required
MISRA C-2012 Rule 15.4 | 1 | Advisory
MISRA C-2012 Rule 17.7 | 4 | Required
MISRA C-2012 Rule 17.8 | 10 | Advisory
MISRA C-2012 Rule 20.5 | 1 | Advisory
MISRA C-2012 Rule 21.6 | 3 | Required
MISRA C-2012 Rule 5.9 | 2 | Advisory
MISRA C-2012 Rule 8.13 | 18 | Advisory
MISRA C-2012 Rule 8.4 | 2 | Required
MISRA C-2012 Rule 8.9 | 7 | Advisory
MISRA C-2012 Rule 9.1 | 1 | Mandatory


</body>
</html>

Test Steps
-----------


**Generate SMP coverity report**
1. Clone the SMP integration [FreeRTOS repo](https://github.com/chinglee-iot/FreeRTOS/tree/smp-integration-coverity-test).
2. cd FreeRTOS/Test/Coverity
3. Update the compiler configuration in Coverity
`cov-configure --force --compiler cc --comptype gcc -- -std=gnu90`
4. Build the (pseudo) application
`cov-build --emit-complementary-info --dir cov-out make multicore1`
or
`cov-build --emit-complementary-info --dir cov-out make multicore2`
5. Go to the Coverity output directory (cov-out) and begin Coverity static analysis
`cd cov-out/`
`cov-analyze --dir . --coding-standard-config ../coverity_misra.config`
6. Generate the report with the following command
`cov-format-errors --dir . --file "Source/tasks.c" --exclude-files '(/Test/|/Source/portable/)' --html-output html-out;`

**Generate single core coverity report**
1. Clone the SMP integration [FreeRTOS repo](https://github.com/chinglee-iot/FreeRTOS/tree/smp-integration-coverity-test).
2. cd FreeRTOS/Test/Coverity
3. Update the compiler configuration in Coverity
`cov-configure --force --compiler cc --comptype gcc -- -std=gnu90`
4. **Build the (pseudo) application ( different make target )**
`cov-build --emit-complementary-info --dir cov-out make singlecore`
5. Go to the Coverity output directory (cov-out) and begin Coverity static analysis
`cd cov-out/`
`cov-analyze --dir . --coding-standard-config ../coverity_misra.config`
6. Generate the report with the following command
`cov-format-errors --dir . --file "Source/tasks.c" --exclude-files '(/Test/|/Source/portable/)' --html-output html-out;`

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
